### ISSUE TITLE
Incorrect orderHistory implementation logic

### ISSUE THREAD

**Issue by franckgarnier21** [DivanteLtd/magento2-rest-client#8]
Hello,

In the file : magento2-rest-client/lib/customers.js

The implementation to retrieve the order history for a customer is not correct.

In the case of guest checkout with email A and a login checkout with email A, when you log in your account, then check the order history, you will see both orders (guest + customer).

The expected behavior is to retrieve only the order history for the current customer.

So I believe you need to add the customer ID in the order search query in order to prevent view order not related to the customer.

Available for more information if needed. I am just doing a POC of VueStoreFront.

**Comment by pkarw** [comment]
Hi! I think its interesting case as it might be natural for users to list their orders (by email) event as made as guests and this is the reason of this implementation :) however I believe we could add an parameter to this call for user to choose the behavior. Can you propose a PR please?

**Comment by rain2o** [comment]
I was about to open an issue for this exact thing. I agree the expected behavior would be to pull orders based on the Customer ID instead of email, for the reason provided in the initial issue description.

A standard Magento instance pulls order history based on the user's ID, as does any e-commerce platform I have used as a customer. I can sympathize with the desired outcome you mentioned @pkarw , however we have discussed these scenarios a lot internally and there are some rare edge-cases where that could actually become a security concern by only pulling by email address. Another scenario where the current logic will cause a problem is when a user changes the email address of their account. Their order history will only show orders with the new email now, losing all previous history. Searching by only customer id will solve this problem.

We have had discussions internally about providing a secure way for customers to attach guest orders to their account after the fact if they desire. This would have to be done in a way where they can prove it's their order. I believe this could solve the problem of users wanting to see all orders they placed no matter their logged-in state. However, I also believe this is a separate issue and should be resolved as a separate feature.

Unfortunately Magento doesn't have an API endpoint for `/orders/me` to pull by user token, and I'm not entirely sure why. Because of this the best solution is search by `customer_id` instead.
